src/clj/puppetlabs/services/master/master_service.clj

```diff
@@ -4,25 +4,31 @@
             [puppetlabs.trapperkeeper.core :refer [defservice]]
             [puppetlabs.services.master.master-core :as core]
             [puppetlabs.puppetserver.certificate-authority :as ca]
-            [puppetlabs.trapperkeeper.services :as tk-services]))
+            [puppetlabs.trapperkeeper.services :as tk-services]
+            [puppetlabs.dujour.version-check :as version-check]))
 
 (defservice master-service
   [[:WebroutingService add-ring-handler get-route]
    [:PuppetServerConfigService get-config]
    [:RequestHandlerService handle-request]
    [:CaService initialize-master-ssl! retrieve-ca-cert! retrieve-ca-crl!]
-   [:JRubyPuppetService]]
+   [:JRubyPuppetService]
+   [:ConfigService get-in-config]]
   (init
    [this context]
    (core/validate-memory-requirements!)
-   (let [path          (get-route this :master-routes)
-         config        (get-config)
-         certname      (get-in config [:puppet-server :certname])
-         localcacert   (get-in config [:puppet-server :localcacert])
-         hostcrl       (get-in config [:puppet-server :hostcrl])
-         settings      (ca/config->master-settings config)
-         jruby-service (tk-services/get-service this :JRubyPuppetService)
-         upgrade-error (core/construct-404-error-message jruby-service)]
+   (let [path              (get-route this :master-routes)
+         config            (get-config)
+         certname          (get-in config [:puppet-server :certname])
+         localcacert       (get-in config [:puppet-server :localcacert])
+         hostcrl           (get-in config [:puppet-server :hostcrl])
+         settings          (ca/config->master-settings config)
+         jruby-service     (tk-services/get-service this :JRubyPuppetService)
+         upgrade-error     (core/construct-404-error-message jruby-service)
+         product-name      (get-in-config [:product :name])
+         update-server-url (get-in-config [:product :update-server-url])]
+
+     (version-check/check-for-updates! {:product-name product-name} update-server-url)
```

Comment:
Do we have a test which validates that the master service makes this call with the right parameters?  If not, it would seem good to create one.  Maybe could stub out the call to `check-for-updates!` with a `with-redefs` so that the full plumbing doesn't have to be exercised for this local unit test?